src/connection.js

```diff
@@ -585,8 +595,12 @@ class Connection extends EventEmitter {
 
   requestTimeout() {
     this.requestTimer = undefined;
-    this.messageIo.sendMessage(TYPE.ATTENTION);
-    return this.transitionTo(this.STATE.SENT_ATTENTION);
+
+    const message = new OutgoingMessage(TYPE.ATTENTION, false, this.messageIo.packetSize(), this.debug);
+    this.messageIo.sendMessage(message);
```

Comment:
Right. My plan is to implement this via a `cancel` method on `OutgoingMessage`, which ends the stream and `_flush` could then handle sending a packet with the attention bit set.